src/Settings.ts

```diff
@@ -82,6 +82,16 @@ export namespace Settings {
         }
     }
 
+    export namespace SettingsFile {
+        export function path(): string | undefined {
+            let config: ConfigurationScope | undefined;
+            if (workspace.workspaceFile !== undefined) {
+                config = workspace.workspaceFile;
+            }
+            return workspace.getConfiguration("maven", config).get<string>("settingsFile");

```

Comment:
> workspaceFile
> 		 * Depending on the workspace that is opened, the value will be:
>		 *  * `undefined` when no workspace or  a single folder is opened
>		 *  * the path of the workspace file as `Uri` otherwise. if the workspace
>		 * is untitled, the returned URI will use the `untitled:` scheme
>		 *

You failed at least two scenarios:
1. for an untitled workspace, `config` would be using untitled scheme, and I don't know if it can correctly fetch **workspace** level settings.
2. for multi-root workspace, it's unable to fetch settings in **workspace folder** level.

Reply:
OK I will change it's scope to global .